### PR TITLE
feat: implement TextInput

### PR DESCRIPTION
**Pull request by TranQuangPhi**
## Features
- TextInput
- TextInput Flat
- TextInput Outlined

## Check List
- [x] Your code builds clean without any errors or warnings.
- [x] Check coding convention.
- [x] Updated Storybook components and stories to reflect any changes made to the UI.
- [x] Test covers all edge cases and coverage is greater than or equal to 80%.

## Proof of Completeness
IOS

https://user-images.githubusercontent.com/105619840/232392866-9af76e8a-2913-4de2-b532-35541b4d1aaa.mov

https://user-images.githubusercontent.com/105619840/232392910-f0993223-8057-49c3-866d-e24175d48789.mov

ANDROID

https://user-images.githubusercontent.com/105619840/232394365-aedc11a1-544f-431f-a2d3-67298d764b70.mov

https://user-images.githubusercontent.com/105619840/232394396-f0a4c5cd-9086-4297-95e8-3cbcbf9a6f59.mov

### CODE REVIEW

File: src/components/Icon/Icon.tsx [hunk 1]
```diff
@@ -0,0 +1,63 @@
+import React from 'react'
+import type {ImageStyle} from 'react-native'
```
Comment (huydosgtech) [inline src/components/Icon/Icon.tsx:2]: duplicate import 'react-native' here

File: src/components/TextInput/TextInput.tsx [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+      clear: () => inputRef.current?.clear(),
+      blur: () => inputRef.current?.blur(),
+    }))
+
+    const handleFocus = useCallback(() => {
+      inputRef.current?.focus()
+    }, [])
+
+    return (
+      <Container style={containerStyle}>
```
Comment (huydosgtech) [inline src/components/TextInput/TextInput.tsx:121]: please fix stylelint on TextInput file

File: src/components/Icon/Icon.tsx (hunk 1 above)
Comment (TranQuangPhi) [reply src/components/Icon/Icon.tsx:2]: The line at row 2 is only import type. I want to separate the import type and the import component into different lines

File: example/src/App.tsx [hunk 3]
```diff
@@ -4,10 +4,44 @@ import StorybookUIRoot from '../.ondevice/Storybook'
...
 const theme = extendTheme({
   colors: {
+    primary: '#7239E5',
+    secondary: '#FFBB33',
+    backgroundPrimary: '#F5F2F1',
+    backgroundSecondary: '#D1D1D1',
+    lightBackground: '#FFFFFF',
+    mainBackground: '#F0F2F3',
+    backgroundColor: '#FFFFFF',
     cardPrimaryBackground: 'green',
```
Comment (hoangSTS) [inline example/src/App.tsx:14]: should we change this to 'cardBackgoundPrimary'?

File: example/src/App.tsx [hunk 4]
```diff
@@ -4,10 +4,44 @@ import StorybookUIRoot from '../.ondevice/Storybook'
...
   },
   darkColors: {
+    primary: '#11EFE8',
+    secondary: '#DADD12',
+    backgroundPrimary: '#F1F1F1',
+    backgroundSecondary: '#6610BC',
+    lightBackground: '#000000',
+    mainBackground: '#F0F2F3',
+    backgroundColor: '#5F6066',
     cardPrimaryBackground: 'gray',
```
Comment (hoangSTS) [inline example/src/App.tsx:34]: same above

**Review by baonguyenhsts — APPROVED** [review]

**Review by vunguyendhSTS — APPROVED** [review]

**Review by huydosgtech — APPROVED** [review]

**Comment by huydosgtech** [comment]
please resolved conflict PR

File: src/components/Icon/Icon.tsx [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+      source={source}
+      style={[
+        {
+          width: size,
+          height: size,
+          tintColor: color,
+        },
+        StyleSheet.flatten(style),
+      ]}
+      resizeMode={resizeMode || 'contain'}
```
Comment (loc-nguyenthien) [inline src/components/Icon/Icon.tsx:60]: We already have default value in line 36

Comment (hoangSTS) [reply src/components/Icon/Icon.tsx]: Done

**Review by chungnguyenhoang — APPROVED** [review]

**Pull request merged**